python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1964 @@
+                visit_id = -1
+
+            donut_table.meta["visit_info"] = {
+                "visit_id": visit_id,
+                "focus_z": float('nan') * u.mm,
+                "boresight_ra": float('nan') * u.deg,
+                "boresight_dec": float('nan') * u.deg,
+                "boresight_rot_angle": float('nan') * u.deg,
+                "boresight_par_angle": float('nan') * u.deg,
+                "boresight_alt": float('nan') * u.deg,
+                "boresight_az": float('nan') * u.deg,
+                "mjd": float('nan'),
+                "donut_radius": float('nan'),
+            }
+
+        # Add detector metadata
+        donut_table.meta["detector"] = detector_name
+        donut_table.meta["camera"] = camera_name
+        donut_table.meta["band"] = band_label
+
+        self.log.debug("Created donut table with %d sources matching standard format", len(donut_table))
+        return donut_table
+
+    def calcZernikesFromExposure(self, exposure: afwImage.Exposure, defocalType: str) -> np.ndarray:
+        """Calculate Zernike coefficients from a single focal position
+        exposure.
+
+        This method processes either an intra-focal OR extra-focal exposure
+        through the TARTS neural network to estimate Zernike coefficients
+        representing wavefront aberrations. The exposure contains donut
+        stamps from one side of the focal plane (not both).
+
+        Parameters
+        ----------
+        exposure : lsst.afw.image.Exposure
+            An exposure containing donut stamps from either the intra-focal
+            OR extra-focal position. This is NOT a full frame exposure
+            with both focal positions - it's specifically one side of
+            the focal pair. Must have valid image data and WCS information.
+
+        Returns
+        -------
+        np.ndarray
+            Array of Zernike coefficients in microns, representing the
+            estimated wavefront aberrations for this focal position.
+            The array length matches the configured Noll indices
+            (default: Z4-Z23).
+
+        Notes
+        -----
+        This method processes one side of the focal pair (intra OR extra).
+        In production, a single "exposure" typically contains both intra
+        and extra-focal donuts in different corners, but this method
+        expects an exposure with donut stamps from only one focal position.
+
+        The exposure should contain donut stamps that the TARTS models
+        are trained to process, not full frame images.
+        """
+        self.log.debug("Calculating Zernikes from exposure; defocalType='%s'", defocalType)
+        with torch.no_grad():
+            pred = self.tarts.deploy_run(exposure)
+        # Convert PyTorch tensor to numpy array
+        pred = self._to_numpy(pred)
+        self.log.debug("TARTS prediction array shape: %s", np.shape(pred))
```

Comment:
What is prediction array contain? Zernikes? It would be good to clarify it in the message